src/emmy/value.cljc

```diff
@@ -438,6 +439,55 @@
        (exact? [_] true)
        (kind [_] goog.math.Long))))
 
+#?(:cljs
+  ;; We find it convenient to be able to decorate "vanilla" JavaScript
+  ;; functions with metadata. In Clojurescript, using `with-meta` on a
+  ;; native function will replace it with an AFn (abstract function)
+  ;; instance, basically an object holding the function and its metadata
+  ;; in two ordinary properties. Such an object does not have the Function
+  ;; prototype, and is therefore not directly callable inJS.
+  ;;
+  ;; To fix this, we allow native functions to store metadata
+  ;; in a symbol-indexed property. The Clojurescript implementation
+  ;; is arguably more "hygienic," and gets around the difficulty of
+  ;; cloning the function as the `IWithMeta` contract would require,
+  ;; but this implementation should be safe enough for our purposes.
+  ;;
+  ;; By using a symbol instead of a string as the key for the metadata,
+  ;; the metadata annotation will not be visible among the object's other
+  ;; normal properties.
+   (let [metadata-symbol (. js/Symbol for "Symbol.__emmy_meta__")]
+     (extend-type function
+       IMeta
+       (-meta [f] (or (gobject/get f metadata-symbol) {})))
+
+    ;; We would like to support IWithMeta, but that interface's contract
+    ;; requires the production of a _new_ object. Instead we provide a function
+    ;; to mutate a native object's metadata in place.
+     (defn set-js-meta!
+       "Mutates the native JS object `o` to have the given metadata. The
+       previous metadata, if any, is discarded. `o` is returned."
+       [o m]
+       (unchecked-set o metadata-symbol m)
+       o))
+   )
+
+#?(:cljs
+   (let [property-name-replacements {"name" "-name"}]
```

Comment:
Speaking from total inexperience...

1. are dashes allowed in JS property names? I know they work for `obj["-name"]` but `obj.-name` is illegal, right? I hit this when realizing that cljs munges `-` to `_`, so if you have `a-x` and `a_x` variables one will shadow the other.
2. I'm not clear on why we need the replacement, so I'll ask if we still need this after the f-name replacement? was that why it was here?

Reply:
I have retired this, my only motive was not to drop a property on the floor that might be useful elsewhere, but that doesn't fix an actual problem we have, so dropped it